### PR TITLE
helm: Initial chart configuration automation.

### LINKED ISSUE #985: Provide Helm chart artifact for installation
#### What would you like to be added:

As part of the releases of this operator, we should also release Helm charts.

#### Why is this needed:

The Security Profiles Operator currently provides two modes of installation:

* Kustomize-generated manifests
* Operator Lifecycle Manager bundle

There are a lot of organizations who depend on Helm to install Kubernetes packages, so it would be ideal to also provide a Helm Chart as means of installation

#### User story covered

All? This would enable more folks to install the Security Profiles Operator through their preferred means.

### PR DESCRIPTION
**Pull request by stephen-fox**
This is a very, very basic helm chart configuration that is auto-generated
by the existing make and kustomize automation. The chart is stored
in `deploy/helm`. The chart is (mostly) generated by executing:
`make deployments`

The `templates/deployment.yaml` file must be manually maintained
as it contains fields that are templated with helm. The other chart files
are either auto-generated by "make" or were created as a part of
`helm create`.

In regards to cert-manager: the project's maintainers do not
support helm's dependency management feature. Speaking with
@JAORMX, we will likely need to document it as a requirement.
Refer to the following URLs for more information:

  - https://discuss.kubernetes.io/t/installing-helm-charts-cert-manager-as-a-dependency-of-my-chart-application/6170
  - https://github.com/cert-manager/cert-manager/pull/2964

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Refer to issue #985.

#### Which issue(s) this PR fixes:
Fixes #985

#### Does this PR have test?
To be decided...?

#### Special notes for your reviewer:
Working with @JAORMX on this :)

#### Does this PR introduce a user-facing change?
Yes - it adds a new deployment method. Refer to TODOs above for more information
about possible documentation.

#### Release note

```release-note
Added a basic helm chart. The chart is generated by running "make deployments"
```

edit: Sorry, GitHub dumped the commit message ontop of the issue template
and I missed editing the template.

### CODE REVIEW

**Comment by saschagrunert** [comment]
Hey @stephen-fox, thank you for this contribution! May I ask you to sign your commits via EasyCLA?

**Comment by stephen-fox** [comment]
Hi @saschagrunert - will do. I am just waiting for approval from my organization.

**Review by JAORMX — COMMENTED** [review]
This is a good start. Let's start looking into having a least a linting job for this, an subsequently a test scenario that uses the helm chart.

**Review by pjbgf — COMMENTED** [review]
@stephen-fox really good stuff, thanks for working on this. :bow: 

It would be great to have a way to ensure the Chart contents does not drift too much from the artefacts within `deploy/`. What are your thoughts on updating Makefile's `verify` target to check that the chart's `crds/operator.yaml` is aligned with the contents of the latest version of `deploy/base/crd.yaml`?

File: charts/security-profiles-operator/Chart.yaml
Comment (saschagrunert) [inline charts/security-profiles-operator/Chart.yaml:6]: We have to update the `release.md` docs that this version has to be bumped as well.

File: charts/security-profiles-operator/values.yaml
Comment (saschagrunert) [inline charts/security-profiles-operator/values.yaml:23]: I think changing those may break the operator, right? So do we wanna make it not configurable?

**Review by saschagrunert — APPROVED** [review]
Great work, I like that!

I think we need a CI job for checking consistency of the helm chart compared to the customize deployment. Not sure how to do that in detail, but we can follow-up on it. 

PTAL @jhrozek @JAORMX @pjbgf

**Comment by stephen-fox** [comment]
Completely agree with @pjbgf and @ccojocar about configuration drift being a problem here.

Let me take a look at the blog Cosmin linked.

**Comment by stephen-fox** [comment]
Unrelated to config drift - it looks like helm does not support updating or deleting CRDs. Is this a problematic for the operator? (https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations)

**Comment by saschagrunert** [comment]
> Unrelated to config drift - it looks like helm does not support updating or deleting CRDs. Is this a problematic for the operator? (https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations)

Yeah we would graduate our APIs to stable slowly which could incorporate a CRD update.

**Comment by saschagrunert** [comment]
@JAORMX @pjbgf I'm kinda fine with the overall approach, what do you think in relation to @stephen-fox's [comment](https://github.com/kubernetes-sigs/security-profiles-operator/pull/1013#issuecomment-1185927152)?

**Review by JAORMX — CHANGES_REQUESTED** [review]
I like the overall approach. I'd say we should also make the Deployment/Pod labels configurable, and it's ready IMO.

**Comment by saschagrunert** [comment]
> @saschagrunert any idea why the olm test broke? Seems to me like it's a legitimate issue.

Yes I think the operator bundle is broken with that change :thinking:

**Comment by stephen-fox** [comment]
Hmm, it's still broken. I'll keep digging into the `make bundle` code.

**Comment by stephen-fox** [comment]
I think I understand at least some of the problem now. The `make bundle` automation `cat`s `deploy/operator.yaml` into the operator-sdk. Since I removed the `Deployment` resource from that file, it ends up missing from the OLM bundle. I am not sure about `clusterserviceversion.yaml`, but I would not be surprised if the missing resource has side effects on that.

**Comment by stephen-fox** [comment]
ps: Changes rebased on `main`.

**Comment by saschagrunert** [comment]
@stephen-fox please rebase again to get the latest CI changes in :)

**Comment by JAORMX** [comment]
This lgtm

**Review by JAORMX — APPROVED** [review]

**Pull request merged**
